vm/systemSmartContracts/esdt.go

```diff
@@ -267,33 +251,204 @@ func (e *esdt) issueToken(owner []byte, arguments [][]byte) error {
 	return nil
 }
 
-func (e *esdt) burn(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	//TODO: implement me
-	return vmcommon.Ok
+func (e *esdt) upgradeProperties(token *ESDTData, args [][]byte) {
```

Comment:
unit tests?

Reply:
will come in next PRs. Want to write full implementation and happy path integration test first.